packages/mobile/src/account/Settings.test.tsx

```diff
@@ -90,4 +99,33 @@ describe('Account', () => {
     )
     expect(tree).toMatchSnapshot()
   })
+
+  it('navigates to PincodeSet screen if entered PIN is correct', async () => {
+    const tree = render(
+      <Provider store={createMockStore({})}>
+        <Settings {...getMockStackScreenProps(Screens.Settings)} />
+      </Provider>
+    )
+    mockedNavigate.mockImplementationOnce(() => {
```

Comment:
Do these tests pass?

The function that we should mock that resolves to a promise with true is `ensurePincode`, not `navigate`. We don't need to provide any behaviour to `navigate`, we need to add this logic to `ensurePincode`.

Also, you're not returning the promise, so this implementation is not returning anything :/

Reply:
been having trouble running it D: i changed the implementation of the mock for `ensurePincode`. does that look correct?